py/picca/delta_extraction/data_catalogues/desi_healpix.py

```diff
@@ -216,35 +230,29 @@ def read_file(self, filename, catalogue):
                 w = np.isnan(spec["FLUX"]) | np.isnan(spec["IVAR"])
                 for key in ["FLUX", "IVAR"]:
                     spec[key][w] = 0.
-                if self.analysis_type == "PK 1D":
-                    if f"{color}_RESOLUTION" in hdul:
-                        spec["RESO"] = hdul[f"{color}_RESOLUTION"].read()
-                    else:
-                        if not reso_from_truth:
-                            self.logger.debug(
-                                "no resolution in files, reading from truth files"
-                            )
-                        reso_from_truth = True
-                        basename_truth = os.path.basename(filename).replace(
-                            'spectra-', 'truth-')
-                        pathname_truth = os.path.dirname(filename)
-                        filename_truth = f"{pathname_truth}/{basename_truth}"
-                        if os.path.exists(filename_truth):
-                            with fitsio.FITS(filename_truth) as hdul_truth:
-                                spec["RESO"] = hdul_truth[
-                                    f"{color}_RESOLUTION"].read()
-                        else:
-                            raise DataError(
-                                f"Error while reading {color} band from "
-                                f"{filename}. Analysis type is 'PK 1D', "
-                                "but file does not contain HDU "
-                                f"'{color}_RESOLUTION'")
+
+                if self.analysis_type != "PK 1D":
```

Comment:
This if feels weird. It makes line 235 be duplicated unnecessarily. How about just checking for `self.analysis_type == "PK 1D"` and then have the block 238 to 247 inside that if?